astropy/utils/iers/iers.py

```diff
@@ -667,29 +667,27 @@ def open(cls):
                 return cls.iers_table
 
         dl_success = False
-        err_list = []
+        err_msg = ''
 
-        for url in all_urls:
-            try:
-                filename = download_file(url, cache=True)
-            except Exception as err:
-                err_list.append(str(err))
-            else:
-                dl_success = True
-                break
+        try:
+            filename = download_file(all_urls[0], sources=all_urls, cache=True)
+        except Exception as err:
+            err_msg = str(err)
+        else:
+            dl_success = True
 
         if not dl_success:
             # Issue a warning here, perhaps user is offline.  An exception
             # will be raised downstream when actually trying to interpolate
             # predictive values.
-            warn(AstropyWarning('failed to download {}, using local IERS-B: {}'
-                                .format(' and '.join(all_urls),
-                                        ';'.join(err_list))))  # noqa
+            warn(AstropyWarning(
+                f'failed to download {" and ".join(all_urls)}, '
+                f'using local IERS-B: {err_msg}'))
             cls.iers_table = IERS_B.open()
             return cls.iers_table
 
         cls.iers_table = cls.read(file=filename)
-        cls.iers_table.meta['data_url'] = str(url)
+        cls.iers_table.meta['data_url'] = all_urls[0]
```

Comment:
Before this patch, the metadata stores the actual URL it is from, but now I don't have that information, so I just store the key of the hash here. Is this a deal breaker? Maybe @aarchiba knows a way around this that I don't.

Reply:
I don't think this is a deal-breaker - the tables are meant to be identical - though given the amount of hassle @taldcroft and I had before figuring out that "IERS B" data in the IERS A files are not, in fact, identical to those of IERS B, and that "final" IERS B data are not, in fact, final, I *think* this was here mostly to be able to find out whether data came from.

Anyway, overall my sense would be to just make the change you have here, but I'd defer to @taldcroft.

Reply:
I'll now just make the decision that this change is OK.